src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIChatClient.cs

```diff
@@ -749,11 +757,41 @@ internal static void ConvertContentParts(ChatMessageContent content, IList<AICon
             _ => new ChatFinishReason(s),
         };
 
+    /// <summary>Sanitizes the author name to be appropriate for including as an OpenAI participant name.</summary>
+    private static string? SanitizeAuthorName(string? name)
+    {
+        if (name is not null)
+        {
+            const int MaxLength = 64;
+
+            name = InvalidAuthorNameRegex().Replace(name, string.Empty);
```

Comment:
Maybe it's just me, but I would rather have this fail with a clear error message as opposed to surreptitiously collapsing the value to something acceptable.